announcements/views.py

```diff
@@ -128,11 +128,38 @@ def get_form_kwargs(self):
 
 class AnnouncementListAPIView(APIView):
     def get(self, request, format=None):
-        announcements = Announcement.objects.this_week()
+        announcements = self.get_queryset()
         serializer = AnnouncementSerializer(announcements, many=True)
         return Response(serializer.data)
 
 
+    def get_queryset(self):
+        time = self.request.query_params.get('time', None)
+        exclude_ongoing = self.request.query_params.get('exclude_ongoing', False)
+        audience = self.request.query_params.get('audience', None)
+
+        queryset = Announcement.objects.all()
+
+        if time is not None:
+            if time == 'time_week':
```

Comment:
Is 'time_week' intentional, or should this be 'this_week'?

Reply:
Double typo... should also be a hyphen.